### PR TITLE
firewall: Show ports that were directly added to the firewall

### PR DESCRIPTION
**Pull request by Gundersanne**
So ports can also directly be added to firewalls. So far this PR doesn't allow you to add or remove them, it just shows them. I'm not sure if we want to support modifying them because feature creep on the firewall page wouldn't really have an end as there's so much we could support.

image

### CODE REVIEW

**Comment by garrett** [comment]
- **Option 1**:
   - Include a list of ports in the zone's header, right after "Interfaces".
   - It could say something like "Included ports" and then list the ports.
- **Option 2**:
   - Add a row in services, probably at the top, and treat it like any typical service.
   - It could be called "Ports included in zone" or something similar.
   - Ports would be in a comma separated list per protocol (TCP/UDP).
   - It wouldn't be removable nor would it be editable. (Unless we want to open up cans of worms — but I 100% agree with your feature creep comment.)

I like option 2 better as it would be more consistent with the rest of the design and kind of similar to what you have already.

File: test/verify/check-networking-firewall [hunk 1]
```diff
@@ -161,6 +169,12 @@ class TestFirewall(NetworkCase):
         m.execute("firewall-cmd --permanent --new-service=empty && firewall-cmd --reload && firewall-cmd --add-service=empty")
         b.wait_present(".zone-section[data-id='public'] tr[data-row-id='empty']")
 
+        if m.image not in ["rhel-8-2-distropkg"]: # Changed in #13686
+            m.execute("firewall-cmd --add-port=9999/udp")
+            b.wait_in_text(".zone-section[data-id='public'] tr[data-row-id='public-ports']", "9999")
+            m.execute("firewall-cmd --remove-port=9999/udp")
+            b.wait_not_present(".zone-section[data-id='public'] tr[data-row-id='public-ports']")
```
Comment (martinpitt) [inline test/verify/check-networking-firewall]: Can you please test a TCP port as well, and a second port to make sure that the formatting and sorting works as intended?

Comment (Gundersanne) [reply test/verify/check-networking-firewall]: sure

**Comment by martinpitt** [comment]
@Gundersanne: On landing, please fix the commit message "to the fiewall" → "to the zone"

**Comment by Gundersanne** [comment]
@garrett Interesting i haven't seen the range before. That can be kinda confusing I guess.

**Comment by garrett** [comment]
Yeah, doing some user testing on the firewall page would help us answer questions like this. In the meanwhile, I hope it's probably good enough, especially since this is probably an edge case.

The one issue I can think of is if someone doesn't want to expose those ports, yet still wants to use the zone for whatever reason. There isn't a way. I guess the answer, right now, is to use a different zone?

I did try to think of different names for the included ports:
- included ports
- included ports in the zone
- included zone ports
- zone ports
- zone port defaults
- default ports
- default zone ports
- ports included in the zone _(I figured this was the clearest, and is what I recommended even though it is a bit verbose)_
- etc.

The only thing I can think of to differentiate it a bit more is to perhaps drop the bolding and make it italicized. Alternatively, we could also consider making the row a lighter shade of grey (except then it would possibly look disabled).

Or just keep it as-is, unless someone has an issue with it... as it's probably fine. Hopefully. :thinking:

**Comment by marusak** [comment]
my two cents:
It really looks like an overview of what ports are enabled by the following services.
I think it could be the last item in the list, the text could be in cursive and it could say that these are additional ports - like _Additional enabled ports_

**Comment by garrett** [comment]
@Gundersanne shared what @marusak's suggestion looks like (via IRC):
additional-ports

I'm very happy with this. It should cut down on any confusion.

One small issue: If there are no services in a zone, then it's kind of odd to have "additional" ones. But I'm guessing zones that have additional ports would also pull in default services. (Is that a fair assumption?) So I'm pretty sure it's a non-issue.

**Pull request merged**